### PR TITLE
Migrate CI Incedents

### PR DESCRIPTION
**Pull request by solker**
To migrate:

Nodes:
- CI Incedent

Messages:
- CI Build _Items_
- CI Incedent _Items_
  - CI Incedent Error
  - CI Incedent Fixed

### CODE REVIEW

**Comment by solker** [comment]
CI Incidents successfully imported except the field `Fixed Build` because this data is empty.
selection_061
selection_058
selection_059
selection_060

**Comment by solker** [comment]
CI Build Items also successfully imported.
selection_065
selection_062
selection_063
selection_064

**Comment by amitaibu** [comment]
:+1:

**Comment by solker** [comment]
CI Incidents Messages migration successfully finished.
selection_073
selection_072
selection_074

**Comment by solker** [comment]
CI Incidents Fixed messages successfully imported. 
selection_078
selection_075
selection_076
selection_077

**Comment by solker** [comment]
@amitaibu everything is migrated. Ready for code review.

File: shoov/modules/custom/shoov_migrate/plugins/message.inc [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+      $ci_build_wrapper = entity_metadata_wrapper('node', $ci_build_node);
+      $arguments = array(
+        '@repo' => $ci_build_wrapper->og_repo->label(),
+        '@branch' => $ci_build_wrapper->field_ci_build->field_git_branch->value(),
+        '@url' => variable_get('shoov_client_url', 'http://localhost:9000') . '/#/ci-incidents/' . $incident_id,
+      );
+      $wrapper->arguments->set($arguments);
+    }
+
+    message_save($msg_object);
```
Comment (amitaibu) [inline shoov/modules/custom/shoov_migrate/plugins/message.inc]: => $msg_object->save();

File: shoov/modules/custom/shoov_migrate/plugins/message.inc [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+
+    if (!$msg_object->mid) {
```
Comment (amitaibu) [inline shoov/modules/custom/shoov_migrate/plugins/message.inc]: remove

File: shoov/modules/custom/shoov_restful/plugins/restful/restful_token_auth/token_auth/access_token/1.0/ShoovGithubAuthAuthentication.class.php [hunk 3]
```diff
@@ -42,7 +42,11 @@ protected function getUser() {
         'client_id' => variable_get('shoov_github_client_id'),
         'client_secret' => variable_get('shoov_github_client_secret'),
         'code' => $request['code'],
+        'X-NewRelic-ID' => 'XQUCV15VGwYJXVhTBAg=',
```
Comment (amitaibu) [inline shoov/modules/custom/shoov_restful/plugins/restful/restful_token_auth/token_auth/access_token/1.0/ShoovGithubAuthAuthentication.class.php]: ?

File: shoov/modules/custom/shoov_migrate/plugins/message.inc [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+   * @return array
+   *  Array of key fields (nid only in this case) of the node that was saved if
+   *  successful. FALSE on failure.
+   */
+  public function import(stdClass $message, stdClass $row) {
+    $msg_object = message_create($this->getBundle(), array('uid' => $message->uid));
+
+    $wrapper = entity_metadata_wrapper('message', $msg_object);
+
+    if ($this->getBundle() == 'ci_build') {
```
Comment (amitaibu) [inline shoov/modules/custom/shoov_migrate/plugins/message.inc]: this is very hardcoded. Lets go over it later

Comment (solker) [inline shoov/modules/custom/shoov_migrate/plugins/message.inc]: I can extend class MigrateDestinationMessage

File: shoov/modules/custom/shoov_restful/plugins/restful/restful_token_auth/token_auth/access_token/1.0/ShoovGithubAuthAuthentication.class.php (hunk 3 above)
Comment (solker) [inline shoov/modules/custom/shoov_restful/plugins/restful/restful_token_auth/token_auth/access_token/1.0/ShoovGithubAuthAuthentication.class.php]: Strange I removed it. Deleted again.

File: shoov/modules/custom/shoov_migrate/plugins/message.inc [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+      $wrapper->field_ci_build->set($message->field_ci_build['destid1']);
+      $wrapper->field_ci_build_status->set(strtolower($message->field_ci_build_status));
+      $wrapper->field_ci_build_timestamp->set(strtotime($message->field_ci_build_timestamp));
+      $wrapper->field_ci_build_log->value = $message->field_ci_build_log;
```
Comment (amitaibu) [inline shoov/modules/custom/shoov_migrate/plugins/message.inc]: `$wrapper->field_ci_build_log->value->set($message->field_ci_build_log);`

**Comment by amitaibu** [comment]
Merged, thanks.

**Pull request closed without merging**